### PR TITLE
[Validator] Deprecate `ValidatorBuilder::enableAnnotationMapping()` and `ValidatorBuilder::disableAnnotationMapping()`

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | Part of https://github.com/symfony/symfony/issues/51381
| License       | MIT
| Doc PR        | -

Required by https://github.com/symfony/symfony/pull/51425

### CODE REVIEW

File: UPGRADE-6.4.md [hunk 1]
```diff
@@ -156,4 +156,6 @@ Validator
  * Deprecate Doctrine annotations support in favor of native attributes
  * Deprecate passing an annotation reader to the constructor signature of `AnnotationLoader`
  * Deprecate `ValidatorBuilder::setDoctrineAnnotationReader()`
- * Deprecate `ValidatorBuilder::addDefaultDoctrineAnnotationReader()`
+ * Deprecate `ValidatorBuilder::addDefaultDoctrineAnnotationReader()
```
Comment (stof) [inline UPGRADE-6.4.md]: missing backtick

File: src/Symfony/Component/Validator/Tests/ValidatorBuilderTest.php [hunk 2]
```diff
@@ -116,9 +116,27 @@ public function testEnableAnnotationMappingWithCustomDoctrineAnnotationReader()
...
+        $this->builder->enableAnnotationMapping();
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testExpectDeprecationWhenDisablingAnnotationMapping()
+    {
+        $this->expectDeprecation('Since symfony/validator 6.4: Method "Symfony\Component\Validator\ValidatorBuilder::disableAnnotationMapping()" is deprecated, use "disableAttributeMapping()" instead.');
+        $this->builder->disableAnnotationMapping();
```
Comment (stof) [inline src/Symfony/Component/Validator/Tests/ValidatorBuilderTest.php:134]: we should not loose the assertion about the fluent API that we had before

File: src/Symfony/Component/Validator/Tests/ValidatorBuilderTest.php [hunk 3]
```diff
@@ -81,7 +81,7 @@ public function testAddMethodMappings()
      */
     public function testEnableAnnotationMappingWithDefaultDoctrineAnnotationReader()
     {
-        $this->assertSame($this->builder, $this->builder->enableAnnotationMapping());
+        $this->assertSame($this->builder, $this->builder->enableAttributeMapping());
```
Comment (stof) [inline src/Symfony/Component/Validator/Tests/ValidatorBuilderTest.php:84]: this test is really about annotation mapping. So I would rather expect both deprecations in it than changing method

File: src/Symfony/Component/Validator/Tests/ValidatorBuilderTest.php [hunk 4]
```diff
@@ -102,7 +102,7 @@ public function testEnableAnnotationMappingWithCustomDoctrineAnnotationReader()
     {
         $reader = $this->createMock(Reader::class);
 
-        $this->assertSame($this->builder, $this->builder->enableAnnotationMapping());
+        $this->assertSame($this->builder, $this->builder->enableAttributeMapping());
```
Comment (stof) [inline src/Symfony/Component/Validator/Tests/ValidatorBuilderTest.php:105]: same here

File: src/Symfony/Component/Validator/ValidatorBuilder.php [hunk 5]
```diff
@@ -187,31 +187,56 @@ public function addMethodMappings(array $methodNames): static
...
+
+    /**
+     * Enables attribute based constraint mapping.
+     *
+     * @return $this
+     */
+    public function enableAttributeMapping(): static
     {
         if (null !== $this->metadataFactory) {
             throw new ValidatorException('You cannot enable annotation mapping after setting a custom metadata factory. Configure your metadata factory instead.');
```
Comment (stof) [inline src/Symfony/Component/Validator/ValidatorBuilder.php:209]: this exception message should be updated

File: src/Symfony/Component/Validator/ValidatorBuilder.php [hunk 6]
```diff
@@ -344,7 +369,7 @@ public function getLoaders(): array
             $loaders[] = new StaticMethodLoader($methodName);
         }
 
-        if ($this->enableAnnotationMapping) {
+        if ($this->enableAttributeMapping) {
             $loaders[] = new AnnotationLoader($this->annotationReader);
```
Comment (stof) [inline src/Symfony/Component/Validator/ValidatorBuilder.php:373]: if you don't rename the loader class as well, this work is incomplete

**Review by stof — CHANGES_REQUESTED** [review]

**Review by xabbuh — CHANGES_REQUESTED** [review]
We also need to update the `FrameworkExtension`.

**Comment by alexandre-daubois** [comment]
Alright, I thought it was the way to deal with the CI that misses some not-yet-merged methods. Closing in favor of https://github.com/symfony/symfony/pull/51425, I addressed your comments there :slightly_smiling_face:

**Pull request closed without merging**